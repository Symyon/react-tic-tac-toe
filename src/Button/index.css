.CustomButton {
    width: 100%;
    background-color: cyan;
    padding: 1rem;
}