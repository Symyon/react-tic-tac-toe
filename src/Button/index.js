import React, { Component } from 'react';
import './index.css';

class Button extends Component {
    render() {
        const { text, onClick } = this.props;
        return (
            <button
                className="CustomButton"
                onClick={onClick}>
                {text}
            </button>
        );
    }
}

export default Button;