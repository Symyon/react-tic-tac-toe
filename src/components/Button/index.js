import React, { Component } from 'react';
import './index.css';

class Button extends Component {
    render() {
        const { text, onClick } = this.props;
        return (
            <div className="CustomButton"
                style={{ backgroundColor: onClick ? 'rgb(157, 167, 111)' : 'rgb(110, 118, 76)', cursor: onClick ? 'pointer' : 'default' }}
                onClick={onClick} >
                {text}
            </div>
        );
    }
}

export default Button;