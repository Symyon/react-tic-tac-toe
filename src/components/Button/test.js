import React from 'react';
import { shallow } from 'enzyme';
import Button from './';

describe('Button', () => {
    it('render correctly without props', () => {
        const component = shallow(<Button />);

        expect(component).toMatchSnapshot();
    })
});