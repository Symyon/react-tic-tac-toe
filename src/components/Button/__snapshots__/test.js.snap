// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render correctly without props 1`] = `
<div
  className="CustomButton"
  onClick={null}
  style={
    Object {
      "backgroundColor": "rgb(110, 118, 76)",
      "cursor": "default",
    }
  }
/>
`;
